src/k8s-engine/progressTracker.ts

```diff
@@ -31,13 +62,10 @@ export default class ProgressTracker {
   }
 
   /**
-   * Run a given action.  The currently running action with the highest priority
-   * will be displayed as the progress.
+   * Register an action.
    * @returns A promise that will be resolved when the passed-in promise resolves.
    */
-  action<T>(description: string, priority: number, promise: Promise<T>): Promise<T>;
-  action<T>(description: string, priority: number, fn: () => Promise<T>): Promise<T>;
-  action<T>(description: string, priority: number, v: Promise<T> | (() => Promise<T>)) {
+  action<T>(description: string, priority: number, v: Promise<T> | (() => Promise<T>)): Promise<T> {
```

Comment:
Please don't drop the overload declarations — that lets TypeScript-aware editors show clear overload options (with useful argument names).
<img width="793" alt="image" src="https://user-images.githubusercontent.com/3977982/145622266-8075908d-2729-402e-b1e1-5a15b2f7651d.png">

Reply:
Ah I see. I didn't understand function overloading properly before this. I'll add these back.